docs/design/usecase/flow-history-get/flow-history-get.md

```diff
@@ -0,0 +1,80 @@
+# Get flow history
+## The idea
+We should be able to get list of the flow operations with timestamps. 
+This information should also include description of the flow actions during the flow operation.
+
+## Implementation
+This feature works in two modes. When the mod is set to `DEFAULT`, we get all the flow history data 
+with the specified parameters. If the `STATUSES` mode is set, we get a list of the flow statuses with timestamps.
+
+![Get flow history](flow-history-get.png "Get flow history sequence diagram")
+
+## API
+Northbound provides following endpoint for getting flow status timestamps:
+
+`GET /flows/:flow_id/history`
+with request params:
+* `flow_id` - Flow ID
+* `time_from` - Start time to search for flow status timestamps (default: 0 (1 January 1970 00:00:00)).
+* `time_to` - End time to search flow status timestamps (default: now).
+* `max_count` - Number of records viewed (default: 100).
+* `mode` - This parameter determines the presentation of the returned data 
+  (`DEFAULT` - all the flow history data, `STATUSES` - only flow statuses with timestamps).
```

Comment:
I don't like this approach, why not to create a separate endpoint like /flows/:flow_id/history/statuses?

Reply:
Agree with you. New separate endpoint looks much better.